skrub/_joiner.py

```diff
@@ -247,31 +248,25 @@ def __init__(
         )
         self.add_match_info = add_match_info
 
-    def _check_dataframe(self, dataframe):
-        # TODO: add support for polars, ATM we just convert to pandas
-        if is_polars(dataframe):
-            return dataframe.to_pandas()
-        if is_pandas(dataframe):
-            return dataframe
-        raise TypeError(
-            f"{self.__class__.__qualname__} only operates on Pandas or Polars"
-            " dataframes."
-        )
-
     def _check_max_dist(self):
         if (
             self.max_dist is None
             or isinstance(self.max_dist, str)
             and self.max_dist == "inf"
         ):
             self.max_dist_ = np.inf
-        else:
+        elif isinstance(self.max_dist, int) or isinstance(self.max_dist, float):
             self.max_dist_ = self.max_dist
+        else:
```

Comment:
I don't think we need this. what do you think someone could mistakenly pass instead of a float and what does the exception look like in that case if we remove the check?

Reply:
I've used a false parameter as if the user copy / pasted the previous argument and it raises `UFuncTypeError: ufunc 'less_equal' did not contain a loop with signature matching types (, ) -> None` on the line `"match_accepted": rescaled_distances <= max_dist`

Removing it as it might be too narrow